src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,846 @@
+
+    # GST for empty circuit or with gates
+    matrix_jk = np.zeros((4**nqubits, 4**nqubits))
+    for k in range(4**nqubits):
+        circ = prepare_states(k, nqubits)
+        if gate is not None:
+            circ.add(gate)
+        for j in range(4**nqubits):
+            new_circ = measurement_basis(j, circ)
+
+            if noise_model is not None and backend.name != "qibolab":
+                new_circ = noise_model.apply(new_circ)
+
+            expectation_val = GST_execute_circuit(
+                new_circ, k, j, nshots, backend=backend
+            )
+            matrix_jk[j, k] = expectation_val
+
+    return matrix_jk
+
+
+def GST_basis_operations(
+    nqubits=None, gjk=None, nshots=int(1e4), noise_model=None, backend=None
+):
+    """Runs gate set tomography for the basis operations. There are 13 basis operations for a single
+        qubit and 241 basis operations for 2 qubits.
+
+        Args:
+        nshots (int, optional): Number of shots used in Gate Set Tomography.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied
+            to simulate noisy computation.
+        gjk (numpy.array): Numpy array with elements Tr(Q_j rho_k) for with no gate.
+            If None, call GST function with gate=None.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+
+        The 13 and 241 basis operations are based on Physical Review Research 3, 033178 (2021).
+    Returns:
+        numpy.array: Numpy array with elements jk equivalent to either Tr(Q_j rho_k) or Tr(Q_j O_l rho_k).
+    """
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    # Implement a check. If nqubits==1, gjk needs to be 4x4 matrix. If nqubits==2, gjk needs to be 16x16 matrix.
+    if gjk is None:
+        gjk = GST(
+            nqubits, gate=None, nshots=nshots, noise_model=noise_model, backend=backend
+        )
+
+    # GST for basis operations.
+    if nqubits is not None:
+        if nqubits == 1:
+            if gjk is None:
+                gjk = GST(nqubits, gate=None, noise_model=noise_model)
+
+            class_qubit_gate = None
+            theta = None
+
+            BasisOps_13 = [
+                gates.I(0).matrix(backend=backend),
+                gates.X(0).matrix(backend=backend),
+                gates.Y(0).matrix(backend=backend),
+                gates.Z(0).matrix(backend=backend),
+                1 / np.sqrt(2) * np.array([[1, 1j], [1j, 1]]),
+                1 / np.sqrt(2) * np.array([[1, 1], [-1, 1]]),
+                1 / np.sqrt(2) * np.array([[(1 + 1j), 0], [0, (1 - 1j)]]),
+                1 / np.sqrt(2) * np.array([[1, -1j], [1j, -1]]),
+                gates.H(0).matrix(backend=GlobalBackend()),
+                1 / np.sqrt(2) * np.array([[0, (1 - 1j)], [(1 + 1j), 0]]),
+                np.array([[1, 0], [0, 0]]),
+                np.array([[0.5, 0.5], [0.5, 0.5]]),
+                np.array([[0.5, -0.5j], [0.5j, 0.5]]),
```

Comment:
I agree with your version of `GST_basis_operations()` except for these matrices to prepare the states `|+>`, `|+y>`, and `|0>`. I think they should be
```suggestion
            gates.H(0).matrix(backend=backend), #|+>
            np.array([[1, 1], [1j, -1j]]) / np.sqrt(2), #|+y>
            gates.I(0).matrix(backend=backend), #|0>
```
 As I understand it, we want to build a channel to prepare those states, and this can be done with reset channels and those gates. In my version of `GST_basis_operations()`, I am using the new matrices

What am I missing?

Reply:
This is a good idea. I will implement this for the gate set tomography of the basis operations in PEC. :)